week3/homework/src/index.js

```diff
@@ -1,3 +1,197 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const express = require('express');
+const fs = require('fs');
+const bodyParser = require('body-parser');
+
+const app = express();
+
+app.use(express.json());
+
+app.use(bodyParser.json());
+app.use(express.urlencoded({ extended: false }));
+
+// Read All Todos
+app.get('/todos', (req, res) => {
+  const db = require('./db');
+  res.status(200).send({
+    msg: 'All todos retrieved successfully',
+    todos: db,
+  });
+});
+
+// Get A Single Todo
+app.get('/todo/:id', (req, res) => {
+  const db = require('./db');
+  const id = parseInt(req.params.id, 10);
+  db.map(todo => {
+    if (todo.id === id) {
+      return res.status(200).send({
+        message: 'todo successfully retrieved',
+        todo,
+      });
+    }
+  });
+  return res.status(404).send({
+    message: `Todo not found`,
+  });
+});
+
+// Create Todo
+app.post('/create/todos', (req, res) => {
```

Comment:
In REST the url should consist of nouns and the http method is the verb that says what we are trying to do with the noun
todo = nound
post = verb means create by convention
The whole idea of REST is to avoid having verbs such as create/add in the url